Can only be used with specific aws-cdk-lib versions due to alpha module peer dependencies

### Version

v1.18.4

### Steps and/or minimal code example to reproduce

The peer-dependencies are specified in a way that is not working when using any cdk version higher than 2.18.0.

```
    "@aws-cdk/aws-apigatewayv2-alpha": "^2.18.0-alpha.0",
    "@aws-cdk/aws-appsync-alpha": "^2.18.0-alpha.0",
    "@aws-cdk/aws-redshift-alpha": "^2.18.0-alpha.0",
    "@aws-cdk/aws-synthetics-alpha": "^2.18.0-alpha.0",
```

These only match `2.18.0-alpha.0`, `2.18.0-alpha.1`, `2.18.0-alpha.2` and so on but not `2.19.0-alpha.0` or in my case `2.22.0-alpha.0`

### Expected behavior

Any minor version higher than 2.18.0-alpha.0 can be used.

### Actual behavior

Error: Error: Declared dependency on version ^2.18.0-alpha.0 of @aws-cdk/aws-apigatewayv2-alpha, but version 2.22.0-alpha.0 was found

### Other details

I couldn't find a way to make it work with overrides or resolutions and to make it even worse there is no notation that matches `2.18.0-alpha.0` and `2.19.0-alpha.0` other than `2.18.0-alpha.0 || 2.19.0-alpha.0`.

Checked the issues so far but it seems I'm forced to include all of the alpha dependencies which are currently sitting inside of the cdk-monitoring package as well even though I'm not using them
```
node_modules/cdk-monitoring-constructs/lib/monitoring/aws-synthetics/SyntheticsCanaryMetricFactory.d.ts(1,24): error TS2307: Cannot find module '@aws-cdk/aws-synthetics-alpha' or its corresponding type declarations.
52
```

After adding this particular dependency I'm not having run time failures because I don't have the 
```
● Test suite failed to run
--
61 |  
62 | Cannot find module '@aws-cdk/aws-redshift-alpha' from 'node_modules/cdk-monitoring-constructs/lib/facade/MonitoringAspect.js'
63 |  
64 | Require stack:
65 | node_modules/cdk-monitoring-constructs/lib/facade/MonitoringAspect.js
66 | node_modules/cdk-monitoring-constructs/lib/facade/MonitoringFacade.js
67 | node_modules/cdk-monitoring-constructs/lib/facade/index.js
68 | node_modules/cdk-monitoring-constructs/lib/index.js
````

is it an expected behavior ?

--- a/.projenrc.ts
+++ b/.projenrc.ts
@@ -80,10 +80,6 @@ _By submitting this pull request, I confirm that my contribution is made under t
 // Experimental modules
 ["@aws-cdk/aws-redshift-alpha"].forEach((dep) => {
   project.deps.addDependency(
-    `${dep}@^${CDK_VERSION}-alpha.0`,
-    DependencyType.PEER
-  );
-  project.deps.addDependency(
     `${dep}@${CDK_VERSION}-alpha.0`,
     DependencyType.DEVENV
   );

--- a/README.md
+++ b/README.md
@@ -29,7 +29,6 @@ In your `package.json`:
     "cdk-monitoring-constructs": "^7.0.0",
 
     // peer dependencies of cdk-monitoring-constructs
-    "@aws-cdk/aws-redshift-alpha": "^2.112.0-alpha.0",
     "aws-cdk-lib": "^2.112.0",
     "constructs": "^10.0.5"
 

--- a/lib/facade/MonitoringAspect.ts
+++ b/lib/facade/MonitoringAspect.ts
@@ -1,4 +1,3 @@
-import * as redshift from "@aws-cdk/aws-redshift-alpha";
 import { IAspect } from "aws-cdk-lib";
 import * as apigw from "aws-cdk-lib/aws-apigateway";
 import * as apigwv2 from "aws-cdk-lib/aws-apigatewayv2";
@@ -17,6 +16,7 @@ import * as kinesisfirehose from "aws-cdk-lib/aws-kinesisfirehose";
 import * as lambda from "aws-cdk-lib/aws-lambda";
 import * as opensearch from "aws-cdk-lib/aws-opensearchservice";
 import * as rds from "aws-cdk-lib/aws-rds";
+import { CfnCluster } from "aws-cdk-lib/aws-redshift";
 import * as s3 from "aws-cdk-lib/aws-s3";
 import * as secretsmanager from "aws-cdk-lib/aws-secretsmanager";
 import * as sns from "aws-cdk-lib/aws-sns";
@@ -341,13 +341,22 @@ export class MonitoringAspect implements IAspect {
 
   private monitorRedshift(node: IConstruct) {
     const [isEnabled, props] = this.getMonitoringDetails(this.props.redshift);
-    if (isEnabled && node instanceof redshift.Cluster) {
+    if (isEnabled && this.isProbablyL2RedshiftCluster(node)) {
+      const cfnCluster = (node as any).cluster as CfnCluster;
       this.monitoringFacade.monitorRedshiftCluster({
-        clusterIdentifier: node.clusterName,
-        alarmFriendlyName: node.node.path,
+        clusterIdentifier: cfnCluster.ref,
+        alarmFriendlyName: cfnCluster.node.path,
         ...props,
       });
     }
+  }
+
+  private isProbablyL2RedshiftCluster(node: IConstruct): boolean {
+    return (
+      (node as any).cluster instanceof CfnCluster &&
+      !!(node as any).clusterName &&
+      !!(node as any).node?.path
+    );
   }
 
   private monitorS3(node: IConstruct) {

--- a/package.json
+++ b/package.json
@@ -65,7 +65,6 @@
     "typescript": "^4.9.5"
   },
   "peerDependencies": {
-    "@aws-cdk/aws-redshift-alpha": "^2.112.0-alpha.0",
     "aws-cdk-lib": "^2.112.0",
     "constructs": "^10.0.5"
   },

